### PR TITLE
[SMALLFIX] Do not mask invariant violations.

### PR DESCRIPTION
**Pull request by jsimsa**

### CODE REVIEW

**Comment by calvinjia** [comment]
LGTM

File: core/common/src/main/java/alluxio/AbstractClient.java [hunk 1]
```diff
@@ -195,11 +195,10 @@ public synchronized void connect() throws IOException, ConnectionFailedException
    */
   public synchronized void disconnect() {
     if (mConnected) {
+      Preconditions.checkNotNull(mProtocol);
```
Comment (aaudiber) [inline core/common/src/main/java/alluxio/AbstractClient.java]: Preconditions.checkNotNull(mProtocol, "mProtocol should never be null when mConnected is true");

Comment (jsimsa) [inline core/common/src/main/java/alluxio/AbstractClient.java]: done

**Comment by apc999** [comment]
+1 LGTM

**Pull request merged**